weaviate/collection/grpc.py

```diff
@@ -412,6 +436,23 @@ def _convert_references_to_grpc_result(
         for name, non_ref_prop in properties.non_ref_properties.items():
             result[name] = non_ref_prop
 
+        for number_array_property in properties.number_array_properties:
+            result[number_array_property.key] = [float(val) for val in number_array_property.vals]
+
+        for int_array_property in properties.int_array_properties:
+            result[int_array_property.key] = [int(val) for val in int_array_property.vals]
+
+        for text_array_property in properties.text_array_properties:
+            result[text_array_property.key] = [str(val) for val in text_array_property.vals]
+
+        for boolean_array_property in properties.boolean_array_properties:
+            result[boolean_array_property.key] = [bool(val) for val in boolean_array_property.vals]
+
+        for uuid_array_property in properties.uuid_array_properties:
+            result[uuid_array_property.key] = [
+                uuid_lib.UUID(val) for val in uuid_array_property.vals
+            ]
```

Comment:
Since there is a way to distinguish a `List[UUID]` in the `proto`, can this also be done for just `UUID` fields?

Reply:
yes good point, I could check the type in weaviate and then write it into an extra field. Could do the same for dates and then we could return both formated/typed

Reply:
will do in another PR